controllers/components/workbenches/workbenches_controller_actions.go

```diff
@@ -0,0 +1,170 @@
+					return err
+				}
+				// If overlay is defined, update paths
+				defaultKustomizePathNbc := "base"
+				if subcomponent.SourcePath != "" {
+					defaultKustomizePathNbc = subcomponent.SourcePath
+				}
+				NotebookControllerPath = filepath.Join(odhdeploy.DefaultManifestPath, "odh-notebook-controller/odh-notebook-controller", defaultKustomizePathNbc)
+
+				rr.Manifests[i].Path = odhdeploy.DefaultManifestPath
+				rr.Manifests[i].ContextDir = "odh-notebook-controller/odh-notebook-controller"
+				rr.Manifests[i].SourcePath = defaultKustomizePathNbc
+			}
+
+			if strings.Contains(subcomponent.ContextDir, "components/notebook-controller") {
+				// Download subcomponent
+				if err := odhdeploy.DownloadManifests(ctx, "odh-notebook-controller/kf-notebook-controller", subcomponent); err != nil {
+					return err
+				}
+				// If overlay is defined, update paths
+				defaultKustomizePathKfNbc := "overlays/openshift"
+				if subcomponent.SourcePath != "" {
+					defaultKustomizePathKfNbc = subcomponent.SourcePath
+				}
+				KfNotebookControllerPath = filepath.Join(odhdeploy.DefaultManifestPath, "odh-notebook-controller/kf-notebook-controller", defaultKustomizePathKfNbc)
+
+				rr.Manifests[i].Path = odhdeploy.DefaultManifestPath
+				rr.Manifests[i].ContextDir = "odh-notebook-controller/kf-notebook-controller"
+				rr.Manifests[i].SourcePath = defaultKustomizePathKfNbc
+			}
+
+			if strings.Contains(subcomponent.URI, DependentComponentName) {
+				// Download subcomponent
+				if err := odhdeploy.DownloadManifests(ctx, DependentComponentName, subcomponent); err != nil {
+					return err
+				}
+				// If overlay is defined, update paths
+				defaultKustomizePath := "overlays/additional"
+				if subcomponent.SourcePath != "" {
+					defaultKustomizePath = subcomponent.SourcePath
+				}
+				NotebookImagesPath = filepath.Join(odhdeploy.DefaultManifestPath, DependentComponentName, defaultKustomizePath)
+
+				rr.Manifests[i].Path = odhdeploy.DefaultManifestPath
+				rr.Manifests[i].ContextDir = DependentComponentName
+				rr.Manifests[i].SourcePath = defaultKustomizePath
+			}
+		}
+	}
+
+	return nil
+}
+
+func createDefaultNotebookNamespace(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	platform := rr.Release.Name
+	if platform == cluster.SelfManagedRhods || platform == cluster.ManagedRhods {
+		// Intentionally leaving the ownership unset for this namespace.
+		// Specifying this label triggers its deletion when the operator is uninstalled.
+		if err := rr.AddResource(&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: cluster.DefaultNotebooksNamespace,
+				Labels: map[string]string{
+					labels.ODH.OwnedNamespace: "true",
+				},
+			},
+		}); err != nil {
+			return fmt.Errorf("failed to add namespace %s to manifests", cluster.DefaultNotebooksNamespace)
+		}
+	}
+
+	return nil
+}
+
+func updatePodSecurityRoleBinding(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
```

Comment:
there is a common function to achieve this: https://github.com/opendatahub-io/opendatahub-operator/blob/feature-operator-refactor/pkg/controller/actions/security/actions.go#L12-L26